src/point-cloud-layer.ts

```diff
@@ -0,0 +1,219 @@
+    getGeometryVector,
+    invertOffsets,
+} from "./utils.js"
+// TODO which accessors are actually needed for a pointcloud layer
+import {
+    GeoArrowExtraPickingProps,
+    computeChunkOffsets,
+    getPickingInfo,
+} from "./picking.js"
+import { ColorAccessor, FloatAccessor, GeoArrowPickingInfo } from "./types.js";
+import { EXTENSION_NAME } from "./constants.js";
+import { validateAccessors } from "./validate.js";
+import { defaultPoolSize } from "threads/dist/master/implementation.browser.js";
+import { Point } from "@geoarrow/geoarrow-js/dist/type.js";
+import { getPointChild } from "@geoarrow/geoarrow-js/dist/child.js";
+
+/* All properties supported by GeoArrowPointCloudLayer */
+export type GeoArrowPointCloudLayerProps = Omit<
+    PointCloudLayerProps<arrow.Table>, // TODO is this still an Arrow Table or is it another datatype? An arrow table is a vector/array like value so is it the same?
+    | "omitted properties here"
+    > &
+    _GeoArrowPointCloudLayerProps &
+    CompositeLayerProps;
+
+/* All properties added by GeoArrowPointCloudLayer */
+type _GeoArrowPointCloudLayerProps = {
+    // data
+    data: arrow.Table,
+
+    /**
+     * If `true`, validate the arrays provided (e.g. chunk lengths)
+     * @default true
+     */
+    _validate?: boolean;
+
+    /**
+     * 
+     * The units of the point size, one of `'meters'`, `'common'`, and `'pixels'`.
+     * @default 'pixels'
+     */
+    sizeUnits?: Unit;
+
+    /**
+     * Global radius of all points, in units specified by `sizeUnits`
+     * @default 10
+     */
+    pointSize: number;
+
+    /**
+     * Material settings for lighting effect.
+     *
+     * @default true
+     * @see https://deck.gl/docs/developer-guide/using-lighting
+     */
+    material?: Material;
+
+    /** 
+    * Center position accessor.
+    * If not provided, will be inferred by finding a column with extension type
+    * `"geoarrow.point"` or `"geoarrow.multipoint"`.
+    */
+    getPosition?: ga.vector.PointVector | ga.vector.MultiPointVector;
+
+    /**
+     * The normal of each object, in `[nx, ny, nz]`.
+     * @default [0,0,1]
+     */
+    getNormal?: 10 // TODO normalAccessor
+    
+    /**
+     * The rgba color is in the format of `[r, g, b, [a]]`
+     * @default [0,0,0,225] 
+     */
+    getColor?: [0,0,0,225] // TODO colorAccessor
+}
+
+// Remove data nd get Position from the upstream default props
+const {
+    data: _data,
+    getPosition: _getPosition,
+    ..._upstreamDefaultProps
+} = PointCloudLayer.defaultProps;
+
+// Default props added by us
+const ourDefaultProps = {
+    _validate: true,
+};
+
+// @ts-expect-error Type error in merging default props with ours
+const defaultProps: DefaultProps<GeoArrowPointCloudLayerProps> = {
+    ..._upstreamDefaultProps,
+    ...ourDefaultProps,
+}
+
+export class GeoArrowPointCloudLayer<
+    ExtraProps extends {} = {},
+    > extends CompositeLayer<Required<GeoArrowPointCloudLayerProps> & ExtraProps>{
```

Comment:
```suggestion
    > extends CompositeLayer<GeoArrowPointCloudLayerProps & ExtraProps>{
```
I made this change recently in https://github.com/geoarrow/deck.gl-layers/pull/94 to fix erroneous typing (in particular, we allow `getPosition` to be not provided, and asserting `Required<>` missed out on some null checks

Reply:
corrected